certlint_test.go

```diff
@@ -62,12 +90,117 @@ func TestTestData(t *testing.T) {
 				continue
 			}
 			for _, err := range result.Errors.List() {
-				fmt.Printf("%s (%s)\n", err.Error(), result.Type)
+				fmt.Printf("Priority: %s, Message: %s (%s)\n", err.Priority(), err.Error(), result.Type)
 			}
 		}
 	}
 }
 
+func TestCLITestData(t *testing.T) {
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	testdata := path.Join(dir, "testdata")
+	files, _ := ioutil.ReadDir(testdata)
+	for _, f := range files {
+		// skip directories
+		if f.IsDir() {
+			continue
+		}
+
+		t.Run(f.Name(), func(t *testing.T) {
+			fmt.Printf("---- CLI %s ----\n", f.Name())
+
+			golden := path.Join(testdata, "golden", f.Name()+".golden")
+			crt := path.Join(testdata, f.Name())
+			bin := path.Join(dir, "certlint")
+
+			// raise errlevel to Alert because non-zero return causes the test to fail
+			cli_args := []string{"-expired", "-errlevel", "alert", "-cert", crt}
+			cmd := exec.Command(bin, cli_args...)
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			//if *update {
```

Comment:
You have some commented cruft here that could be deleted.

Reply:
Ahh yes, I had that there to remind myself to implement something for programatically updating the `golden` files. I'll switch that to a `TODO` that matches the format seen in the other tests.